File: cowork/ClientApp/src/app/models/RoomBooking.ts
import {DateTime, Duration} from "luxon";
import {Room} from './Room';
import {User} from './User';

export class RoomBooking {
    public Id: number;
    public Start: DateTime;
    public End: DateTime;
    public RoomId: number;
    public ClientId: number;
    public Room: Room;
    public Client: User;
}

File: cowork.domain/RoomBooking.cs
using System;

namespace cowork.domain {

    public class RoomBooking {

        public RoomBooking() { }


        public RoomBooking(long id, DateTime start, DateTime end, long roomId, long clientId) {
            Id = id;
            Start = start;
            End = end;
            RoomId = roomId;
            ClientId = clientId;
        }
        
        public RoomBooking(DateTime start, DateTime end, long roomId, long clientId) {
            Id = -1;
            Start = start;
            End = end;
            RoomId = roomId;
            ClientId = clientId;
        }


        public long Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public long RoomId { get; set; }
        public long ClientId { get; set; }

        public User Client { get; set; }
        public Room Room { get; set; }

    }

}